In [113]:
###---<MAIN_FILE>---###      
%load_ext autoreload
%autoreload 2

The autoreload extension is already loaded. To reload it, use:
  %reload_ext autoreload


In [114]:
#-<this_file imports>-#
import glob
import numpy as np
import os
#-<others>-#
import time
import pdb
#-<custom imports>-#
from params.data_params import *         #<ME>
import noise_addition_pipeline           #<OCODE>

In [115]:
#print_parameters()
noise_gen_dir
list_of_candels_image

['/notebooks/CLUMPS_VELA/noise_stamps/bk_v_2.fits',
 '/notebooks/CLUMPS_VELA/noise_stamps/bk_v_1.fits']

In [116]:
''' 
1. Here, we query the list of real images provided by the user, which will be used to construct a real-noise mosaic.
2. Please make sure to change the FULL PATH. Create a folder where you can put some real CANDELS images and refer that path here.
'''
#noise_gen_dir                          #<ME>
#list_of_candels_image                  #<ME>
'''
1. The VELA runs use the labels -- VELA01, VELA02, etc.
2. One can loop through all the VELA runs (01 through 34) in one go. However, i decided to do it one-by-one for double checking purposes.
'''

'\n1. The VELA runs use the labels -- VELA01, VELA02, etc.\n2. One can loop through all the VELA runs (01 through 34) in one go. However, i decided to do it one-by-one for double checking purposes.\n'

In [117]:
################ SB00 (noise-less) images location ###################
'''
# NOTE, NO leading slash at the end of the path...
# NOTE does not include the folder path to each specific folder...
# each SB00 folder generally follows VELAID_F160W_SB00/
# each folder has files named as *_*_*_cam*_WFC3-F160W_SB00.fits
# Change the file naming convention here in the code to whatever the user wishes to run it on..
# e.g., F125W for J-band... ACS for acs bands... etc..
'''

#sb00_loc                               

'\n# NOTE, NO leading slash at the end of the path...\n# NOTE does not include the folder path to each specific folder...\n# each SB00 folder generally follows VELAID_F160W_SB00/\n# each folder has files named as *_*_*_cam*_WFC3-F160W_SB00.fits\n# Change the file naming convention here in the code to whatever the user wishes to run it on..\n# e.g., F125W for J-band... ACS for acs bands... etc..\n'

In [118]:
############ Fits save file destination directory for noise-added images #############
# NOTE, NO leading slash at the end of the path...

#destin_dir                 # destin_dir = '/Users/km4n6/Desktop/temp/noise_added'


In [102]:
################# Plot save location #################
# plot_save_loc = '/Users/km4n6/Dropbox/bharath/Mantha Method/plots/CANDELS_Tidal_Feature_Identification/CAN_noise_stamp_generation'

#plot_save_loc            # plot_save_loc = '/Users/km4n6/Desktop/temp/plots'
try:
  os.stat(plot_save_loc)
except:
  os.mkdir(plot_save_loc)
#i.e. <ME> Podemos hacer una lista con los id de galaxias

In [122]:
###---<MAIN_LOOP>---### (CORE) #<OCODE>  # Going through each VELA run in the following for loop.
time_start = time.time()
VELA_number = ['01']                #VELA_id  #id_gal for the loop
for each_vela_run in VELA_number:                          #<ME> uso de "ID_list"(data_params.py)  #['01']
    # If a destination folder where the noise-added images need to be placed does not exist, then make the folder with corresponding VELA run number.
    if not os.path.exists('%s/VELA%s'%(destin_dir,each_vela_run)):
        os.mkdir('%s/VELA%s'%(destin_dir,each_vela_run))
    
    # Get the list of noise-less vela images matching the VELA run ID number. The asterisk means that search for all camera angles and timesteps.
    # Please change the filter name and the total path as needed.
    list_of_noise_less_vela_image = glob.glob('%s/VELA%s_F160W_SB00/*_*_*_cam*_WFC3-F160W_SB00.fits'%(noise_dir,each_vela_run))  #<ME> (OCODE sb00_loc)
    
    # For each vela image in the list,
    for vela_image in list_of_noise_less_vela_image:
        can_image = np.random.choice(list_of_candels_image) 
        
        noise_addition_pipeline.extract_noise_and_exp_patches(can_image,                
                                                              vela_image,
                                                              make_plots    = False,
                                                              is_image_cube = False,
                                                              plots_loc     = plot_save_loc,
                                                              fits_save_loc = destin_dir)       #<ME> make_plots=False, is_image_cube=False
        
        print('Added noise to %s\n--------\n'%os.path.basename(vela_image))
        
time_end = time.time()

/usr/local/lib/python3.4/site-packages/matplotlib/axes/_axes.py:6462: UserWarning: The 'normed' kwarg is deprecated, and has been replaced by the 'density' kwarg.
  warnings.warn("The 'normed' kwarg is deprecated, and has been "
/usr/local/lib/python3.4/site-packages/astropy/visualization/mpl_normalize.py:117: RuntimeWarning: invalid value encountered in true_divide
  np.true_divide(values, self.vmax - self.vmin, out=values)
/usr/local/lib/python3.4/site-packages/matplotlib/colors.py:504: RuntimeWarning: invalid value encountered in less
  xa[xa < 0] = -1


Added noise to VELA01_a0.130_sunrise_cam00_WFC3-F160W_SB00.fits
--------

Added noise to VELA01_a0.380_sunrise_cam00_WFC3-F160W_SB00.fits
--------

Added noise to VELA01_a0.230_sunrise_cam00_WFC3-F160W_SB00.fits
--------

Added noise to VELA01_a0.110_sunrise_cam00_WFC3-F160W_SB00.fits
--------

Added noise to VELA01_a0.410_sunrise_cam00_WFC3-F160W_SB00.fits
--------

Added noise to VELA01_a0.360_sunrise_cam00_WFC3-F160W_SB00.fits
--------

Added noise to VELA01_a0.440_sunrise_cam00_WFC3-F160W_SB00.fits
--------

Added noise to VELA01_a0.300_sunrise_cam00_WFC3-F160W_SB00.fits
--------

Added noise to VELA01_a0.290_sunrise_cam00_WFC3-F160W_SB00.fits
--------

Added noise to VELA01_a0.490_sunrise_cam00_WFC3-F160W_SB00.fits
--------

Added noise to VELA01_a0.060_sunrise_cam00_WFC3-F160W_SB00.fits
--------

Added noise to VELA01_a0.190_sunrise_cam00_WFC3-F160W_SB00.fits
--------

Added noise to VELA01_a0.430_sunrise_cam00_WFC3-F160W_SB00.fits
--------

Added noise to VELA01_a0.280_sunrise_c

In [112]:
print("execution_time:",time_end-time_start)
print(VELA_id)
print(VELA_number[0])

execution_time: 0.004202365875244141
02
02


#INFO#
https://docs.scipy.org/doc/numpy/reference/generated/numpy.hstack.html

#-CHANGES-# in noise_addition_pipeline.py
28/03 - Changed the data HDU index inside fits from [1] to [0]. (PrimaryHDU)
28/03 - Changed "how_many_hstacks = int(x_shape/10.)" y "how_many_vstacks = int(y_shape/10.)" ahora son enteros
28/03 - /usr/local/lib/python3.4/site-packages/matplotlib/axes/_axes.py:6462: UserWarning: The 'normed' kwarg is deprecated, and has been replaced by the 'density' kwarg.